trieste/utils/pareto.py

```diff
@@ -148,3 +148,42 @@ def hypervolume_indicator(self, reference: TensorType) -> TensorType:
         hypervolume = tf.reduce_sum(tf.reduce_prod(upper - lower, 0))
 
         return tf.reduce_prod(reference[None] - min_pfront) - hypervolume
+
+    def get_hypercell_bounds(self, anti_reference: TensorType, reference: TensorType) -> Tuple:
+        """
+        Get the partitioned hypercells lower and upper bounds
+        :param anti_reference: a worst point to use with shape [D].
+            Defines the lower bound of the hypercell
+        :param reference: a reference point to use, with shape [D].
+            Defines the upper bound of the hypervolume.
+            Should be equal or bigger than the anti-ideal point of the Pareto set.
+            For comparing results across runs, the same reference point must be used.
+        :return: lower, upper bounds or the partitioned cell
+        :raise ValueError (or `tf.errors.InvalidArgumentError`): If ``reference`` has an invalid
+            shape.
+        """
+        tf.debugging.assert_greater_equal(reference, self.front)
+        tf.debugging.assert_greater_equal(self.front, anti_reference)
+
+        tf.debugging.assert_shapes(
+            [
+                (self.bounds.lower_idx, ["N", "D"]),
+                (self.bounds.upper_idx, ["N", "D"]),
+                (self.front, ["M", "D"]),
+                (reference, ["D"]),
+                (anti_reference, ["D"]),
+            ]
+        )
+
+        pseudo_pfront = tf.concat((anti_reference[None], self.front, reference[None]), axis=0)
+        N = tf.shape(self.bounds.upper_idx)[0]
+        D = tf.shape(self.bounds.upper_idx)[1]
```

Comment:
```suggestion
    N, D = tf.shape(self.bounds.upper_idx)
```

Reply:
Not very sure if it is proper to mention here, but it seems when I used the suggested more compact notation, it will raise an issue when running the notebook of analytic EHVI: 
```
    OperatorNotAllowedInGraphError: iterating over `tf.Tensor` is not allowed: AutoGraph did convert this function. This might indicate you are trying to use an unsupported feature.
```
I do not have an idea on this, maybe any comments?

Reply:
oh wow, that's odd. Especially since tuple unpacking defers to `__iter__`, and so does `for`, and `for` is supported by `tf.function`

Reply:
yeah just stick with your version